core/src/json_schema.rs

```diff
@@ -282,6 +363,297 @@ pub mod simplified {
 
 #[cfg(test)]
 mod unit {
+    macro_rules! remove_multiple_test {
+        ($func_name:ident, expected = $expected:expr, input = $input:expr) => {
+            #[test]
+            fn $func_name() {
+                let res = $input.remove_multiple().unwrap();
+
+                assert_eq!(res, $expected);
+            }
+        };
+    }
+
+    // Testing TryFrom<json_schema::Schema> for json_schema::simplified::Atomic
+    mod remove_multiple {
+        use crate::{
+            json_schema::{BsonType, BsonTypeName, Items, Schema},
+            map,
+        };
+
+        remove_multiple_test!(
+            multiple_atomic,
+            expected = Schema {
+                bson_type: None,
+                properties: None,
+                required: None,
+                additional_properties: None,
+                items: None,
+                any_of: Some(vec![
+                    Schema {
+                        bson_type: Some(BsonType::Single(BsonTypeName::Int)),
+                        properties: None,
+                        required: None,
+                        additional_properties: None,
+                        items: None,
+                        any_of: None,
```

Comment:
```suggestion
                        ..Default::default()
```
[nit] we could make these tests a marginally more readable by shrinking them in this way wherever we just want to set one or two fields and make the rest `None` -- up to you if you think it is worth the change!